components/analytics/components/GeoDistribution/index.js

```diff
@@ -30,7 +31,18 @@ class GeoDistribution extends React.Component {
 	};
 
 	componentDidMount() {
-		const { fetchAppGeoDistribution } = this.props;
+		const { fetchAppGeoDistribution, filterId, selectFilterValue, filters } = this.props;
+		const urlParams = getUrlParams(window.location.search);
+		if (
+			urlParams.from &&
+			urlParams.to &&
+			filters.from !== urlParams.from &&
+			filters.to !== urlParams.to
+		) {
+			selectFilterValue(filterId, 'from', urlParams.from);
+			selectFilterValue(filterId, 'to', urlParams.to);
+			return;
+		}
```

Comment:
@jyash97 You're repeating the line no. 35 to 45 in each component. We do things in a way that it is easy to maintain. You can create a Parent component to update the filters in `redux`.

Reply:
So what that Parent Component will do? 

The issue is that every analytics page is having different fetch API ( Not action but API calls ), and I will need to add the `if` condition in all Components because some analytics pages use redux some are not using it so it is important to have that condition.

Reply:
@jyash97 The parent component will just set the filter.
```js
const urlParams = getUrlParams(window.location.search);
		const { filterId, selectFilterValue, filters } = this.props;
		if (
			urlParams.from &&
			urlParams.to &&
			filters.from !== urlParams.from &&
			filters.to !== urlParams.to
		) {
			selectFilterValue(filterId, 'from', urlParams.from);
			selectFilterValue(filterId, 'to', urlParams.to);
			return;
		}
```

Reply:
I will have this condition in all didMount to prevent 2 API Calls - 

```
if (
			urlParams.from &&
			urlParams.to &&
			filters.from !== urlParams.from &&
			filters.to !== urlParams.to
		) {
			return;
		}
```